subiquity/server/server.py

```diff
@@ -533,6 +532,48 @@ async def start_site(self, runner: web.AppRunner):
     def base_relative(self, path):
         return os.path.join(self.base_model.root, path)
 
+    def load_cloud_config(self):
+        # cloud-init 23.3 introduced combined-cloud-config, which helps to
+        # prevent subiquity from having to go load cloudinit modules.
+        # This matters because a downgrade pickle deserialization issue may
+        # occur when the cloud-init outside the snap (which writes the pickle
+        # data) is newer than the one inside the snap (which reads the pickle
+        # data if we do stages.Init()).  LP: #2022102
+
+        # The stages.Init() code path should be retained until we can assume a
+        # minimum cloud-init version of 23.3 (when Subiquity drops support for
+        # Ubuntu 22.04.2 LTS and earlier, presumably)
+
+        cloud_cfg = get_host_combined_cloud_config()
+        if len(cloud_cfg) > 0:
+            log.debug("loading cloud config from combined-cloud-config")
+            system_info = cloud_cfg.get('system_info', {})
+            default_user = system_info.get('default_user', {})
+            self.installer_user_name = default_user.get('name')
```

Comment:
I think this makes sense, the only thing I can think is a little advertised user-data can manipulate what the default user is using deprecated keys that could override the configuration present in system_info. 

Our example doc on this is below and [in doc examples for users and groups](https://cloudinit.readthedocs.io/en/latest/reference/modules.html#users-and-groups)
```
        user:
          name: mynewdefault
          sudo: null
```

I'll think through this path when the PR is up as this #cloud-config override probably also affects the other logical path you have with stages.Init() too.

Reply:
If this is the combined-cloud-config version of `ug_util.extract_default`, then I think we're set.

The `None` [default user](https://github.com/canonical/subiquity/pull/1687) one was rough because it could be encountered with certain cloud-config before we even got as far as showing the UI.

My goal isn't some sort of comprehensive support for weird corner cases here - more along the lines of "don't crash".